src/Cache/CacheEngine.php

```diff
@@ -279,13 +280,40 @@ abstract public function decrement(string $key, int $offset = 1);
      */
     abstract public function delete($key): bool;
 
+    /**
+     * Delete a key from the cache asynchronously
+     *
+     * Empty method and always raises an exception.
+     * It needs to be implemented in a concrete cache engine.
+     *
+     * @param string $key Identifier for the data
+     * @return bool True if the value was successfully deleted, false if it didn't exist or couldn't be removed
+     */
+    public function deleteAsync($key): bool
+    {
+        throw new NotImplementedException('There is no implementation of a concrete cache engine.');
```

Comment:
```suggestion
        throw new NotImplementedException('The deleteAsync() method is not supported by' . static::class . '. Used delete() instead.');
```

We can like to give more direction in our error messages where possible.